### DISCUSSION TITLE
I've been playing around with BlueSCSI, and I love it. then it got me thinking.

### DISCUSSION THREAD

**Discussion by johnkgibson** [BlueSCSI/BlueSCSI-v2#192] [General]
I discovered another project called Piscsi, which uses a whole Raspberry Pi to interface SCSI ports with USB peripherals. I understand that BlueSCSI runs on the Pico, but could the code base be modified to run on a Raspberry Pi along with PISCS?

I have recently been put into semi-retirement and have decided that I now have time to get back to coding and just screwing around. Does anyone have any thoughts or ideas on how this might be done, or has my brain dreamed up a crazy scheme? I would love to know.

The issue is that I have many vintage computers that have SCSI and booting from a MicroSD is great, but I would like the ability to boot from USB hard, and flash drives. 

Thanks for listening to my rantings!

**Comment by androda** [comment]
The BlueSCSI codebase does practically the same thing as PiSCSI in general.  They both emulate SCSI devices like hard drives, CDROMs, and both can emulate the DaynaPort SCSI to Ethernet converter.

However I don't believe either BlueSCSI or PiSCSI presently act as a USB to SCSI adapter.  Is that what you're looking for?

**Reply by johnkgibson** [reply]
Welp here we are on the interent and I assumed something that wasn't real. However, my PiSCSI will be on a Raspberry Pi 4 so now I am thinking if there would be a way to bridge the SCSI to the Pi's USB ports to allow USB hard drives to be added to the Scsi bus and have the ability to boot from them.

**Reply by androda** [reply]
> allow USB hard drives to be added to the Scsi bus and have the ability to boot from them.

PiSCSI might be able to do that, but I think it's intended to be used with drive image files like BlueSCSI.  Would be an interesting feature to hook up an HFS+ drive to USB and directly mount that to the SCSI bus.

**Comment by erichelgeson** [comment]
Since PiSCSI runs linux, and everything on linux is a file (even USB devices) you _can_ mount a USB thumb drive on PiSCSI _and_ serve it through to SCSI. Of course the vintage scsi host using the device would need to understand the file system on the USB drive (probably FAT16 is the most universal)

USB Dvrive -connected to-> PiSCSI -> `ln -s /dev/<device> /home/pi/images/USB-Drvie.hdr` -> SCSI -> VintageMachine

Then you can mount the `USB-Drive.hdr` via PiSCSI/PiSCSI web and it will show up on your vintage machine. I've done this with a USB Zip drive just for fun, but will apply to any mass storage device.

**Reply by stinkerton18** [reply]
If you don't know what /dev/<device> to use, you can use the lssscsi utility to list all USB connected devices like so:

1. Login to the PiSCSI either using a directly connected keyboard/monitor or through SSH
2. Install lsscsi with: `sudo apt-get install lsscsi`
3. Run lsscsi to see what /dev/<device> was given to the Zip drive

In this case on mine, the Zip 250 was added as /dev/sda
![2024-09-19_12-59-38](https://github.com/user-attachments/assets/ce69e61d-a73d-4518-acd3-720c5ef5d91f)

**Reply by johnkgibson** [reply]
Okay, I will work this out in a couple of days and let everyone know how it goes.
